src/main.rs

```diff
@@ -0,0 +1,18 @@
+pub mod api_client;
+pub mod error;
+pub mod serde;
+pub mod types;
+use api_client::Client;
+use tokio::main;
+use url::{ParseError, Url};
```

Comment:
I don't think we need `ParseError` here
```suggestion
use url::Url;
```